### PR TITLE
[13.0] Create module website_event_sale_single_attendee

### PR DESCRIPTION
**Pull request by grindtildeath**
This module allows to define an event with 'Single Attendee Registration' so
that when a user registers on the website, he will not have to enter a name and
an email for each ticket, but only one name and one email, that will be used
as attendee for every ticket purchased.

### CODE REVIEW

**Review by chienandalu — COMMENTED** [review]
Maybe you could be interested in a website extension to this one: https://github.com/OCA/event/tree/12.0/event_sale_registration_multi_qty

**Comment by grindtildeath** [comment]
@chienandalu I just gave it a look, but in our case, the issue as I see it, is to rely on `event_registration_multi_qty` to add the `qty` field on `event.registration`.

I'm not sure what's the best "all around" solution though, between having a quantity field on `event.registration` or generating mutliple `event.registration` for the same attendee as Odoo and this module does...

**Comment by chienandalu** [comment]
I'm not sure either :sweat_smile: Your purpose does it as the standard `sale.order` does it. With multi_qty you'd be stuck to those modules, although maybe you'd get rid of side issues like the one with surveys...

Your choice :)

File: website_event_sale_single_attendee/readme/ROADMAP.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* This module breaks the feature of website_event_questions, and a glue module
+  must restrict the checkbox 'Ask each attendee' (`event.question.is_individual`)
+  to False, if the event is set with 'Single Attendee Registration'.
```
Comment (grindtildeath) [inline website_event_sale_single_attendee/readme/ROADMAP.rst:3]:
```suggestion
  to False, if the event is set with 'Single Attendee Registration'.
* As only one e-mail is registered, the attendee will receive as many e-mails as tickets he ordered.
```

File: website_event_sale_single_attendee/models/event.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    single_attendee_registration = fields.Boolean(
+        help="Check this box to ask for a single attendee at registration"
+    )
+
+    @api.onchange('event_type_id')
+    def _onchange_type(self):
+        res = super()._onchange_type()
+        if self.event_type_id and self.event_type_id.single_attendee_registration:
+            self.single_attendee_registration = self.event_type_id.single_attendee_registration
+        return res
```
Comment (mmequignon) [inline website_event_sale_single_attendee/models/event.py:19]: Why not a related field ?

File: website_event_sale_single_attendee/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                    ticket_id = key.split("-", 1)[1]
+                    ticket_ids_quantities[ticket_id] = value
+                else:
+                    registration_details[key] = value
+            # Multiply registration details for each ticket according to the
+            # quantities
+            if ticket_ids_quantities:
+                registrations = []
+                for ticket_id, qty in ticket_ids_quantities.items():
+                    for cnt in range(0, int(qty)):
```
Comment (mmequignon) [inline website_event_sale_single_attendee/controllers/main.py:28]:
```suggestion
                    for __ in range(int(qty)):
```

**Review by i-vyshnevska — APPROVED** [review]

**Review by mmequignon — COMMENTED** [review]
Didn't test it, but code looks good.

File: website_event_sale_single_attendee/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+{
+    "name": "Website Event Sale Single Attendee",
+    "summary": "Register a single attendee for multiple tickets",
+    "version": "12.0.1.0.0",
```
Comment (grindtildeath) [inline website_event_sale_single_attendee/__manifest__.py:6]:
```suggestion
    "version": "13.0.1.0.0",
```

**Pull request closed without merging**